src/main/php/xp/web/srv/Output.class.php

```diff
@@ -51,6 +52,15 @@ public function begin($status, $message, $headers) {
    * @return void
    */
   public function write($bytes) {
-    $this->socket->write($bytes);
+    try {
+      $this->socket->write($bytes);
```

Comment:
Should be use `fwrite($this->socket->getHandle())` and check for it returning `false` instead of using the more heavy-weight exception mechanism?

Reply:
Nope, the code doesn't get better that way.